### PR TITLE
Consolidating test statics

### PR DESCRIPTION
**Pull request by trwalke**
# Consolidating test statics

- [x] You've read the [Contributor Guide](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/Contributing.md) and [Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
- [x] You've included unit or integration tests for your change, where applicable.
- [x] You've included inline docs for your change, where applicable.
- [x] If any gains or losses in performance are possible, you've included benchmarks for your changes. [More info](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/wiki/Benchmarking-in-Wilson)
- [x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

Summary of the changes (Less than 80 chars)

## Description

The issue here is that there are 2 statics that have the same data and are supposed to be kept in sync. However, this is quite tedious and if a change is made unknowingly to one of the static pairs, it can be challenging to find the issue when tests fail during comparisons. This PR attempts to consolidate the static pairs into one source and the different forms of the statics can be acquired though subsequent properties.

Fixes #[2272](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2272)

### CODE REVIEW

File: test/Microsoft.IdentityModel.TestUtils/Default.cs [hunk 1]
```diff
@@ -8,6 +8,7 @@
 using System.Security.Claims;
 using System.Security.Cryptography;
 using System.Security.Cryptography.X509Certificates;
+using System.Xml.Linq;
```
Comment (msbw2) [inline test/Microsoft.IdentityModel.TestUtils/Default.cs]: Why is this added?
```suggestion
```

Comment (trwalke) [reply test/Microsoft.IdentityModel.TestUtils/Default.cs]: Not needed. they are automatically added by the IDE when writing code

File: test/Microsoft.IdentityModel.TestUtils/ReferenceTokens.cs [hunk 2]
```diff
@@ -373,39 +373,32 @@ public static string SamlToken_Formated
...
+            get => new JObject()
             {
                 { JwtRegisteredClaimNames.Email, "Bob@contoso.com" },
                 { JwtRegisteredClaimNames.GivenName, "Bob" },
                 { JwtRegisteredClaimNames.Iss, Issuer },
                 { JwtRegisteredClaimNames.Aud, Audience },
                 { JwtRegisteredClaimNames.Iat, EpochTime.GetIntDate(IssueInstant).ToString() },
                 { JwtRegisteredClaimNames.Nbf, EpochTime.GetIntDate(NotBefore).ToString()},
-                { JwtRegisteredClaimNames.Exp, EpochTime.GetIntDate(Expires).ToString() }
+                { JwtRegisteredClaimNames.Exp, EpochTime.GetIntDate(Expires).ToString() },
```
Comment (trwalke) [inline test/Microsoft.IdentityModel.TestUtils/ReferenceTokens.cs:390]:
```suggestion
                { JwtRegisteredClaimNames.Exp, EpochTime.GetIntDate(Expires).ToString() }
```

**Comment by jennyf19** [comment]
Looks like the tests are failing do they work locally?

**Comment by trwalke** [comment]
> Looks like the tests are failing do they work locally?

Yes they worked locally, I think it is just a flaky test run.
image

@jennyf19

**Comment by jennyf19** [comment]
> > Looks like the tests are failing do they work locally?

hmmm...I thought we were trying to remove the flakiness? 🎃

File: test/Microsoft.IdentityModel.TestUtils/Default.cs [hunk 3]
```diff
@@ -859,20 +829,37 @@ public static List<Claim> SamlClaims
...
-                { ClaimTypes.HomePhone, "555.1212" },
-                { ClaimTypes.Role, new List<string>{"Developer", "Sales" } },
-                { ClaimTypes.StreetAddress, "123AnyWhereStreet\r\nSomeTown/r/nUSA" },
-                { ClaimsIdentity.DefaultNameClaimType, "Jean-S�bastien" }
-            };
+                Dictionary<string, object> dictionary = new Dictionary<string, object>();
+
+                foreach (Claim claim in SamlClaims)
+                {
+                    if (dictionary.ContainsKey(claim.Type))
```
Comment (jennyf19) [inline test/Microsoft.IdentityModel.TestUtils/Default.cs:842]: nit: use `TryGetValue` here to avoid a double lookup

**Review by jennyf19 — APPROVED** [review]
Left a small comment, non-blocking

**Comment by trwalke** [comment]
> > > Looks like the tests are failing do they work locally?
> 
> hmmm...I thought we were trying to remove the flakiness? 🎃

Oh for sure, but that flakiness is coming from somewhere else in the code/build infrastructure.

**Review by msbw2 — APPROVED** [review]

**Review by kllysng — APPROVED** [review]
Thanks Travis
